src/ui/public/state_management/state_storage/hashing_store.js

```diff
@@ -49,25 +42,12 @@ class HashingStore {
    *  @return {string} the hash of the value
    */
   hashAndSetItem(object) {
-    const json = angular.toJson(object);
+    const json = JSON.stringify(object);
     const hash = this._getShortHash(json);
     this._store.setItem(hash, json);
     return hash;
   }
 
-  // private api
-
-  /**
-   *  calculate the full hash of a json object
-   *
-   *  @private
-   *  @param {string} json
-   *  @return {string} hash
-   */
-  _createHash(json) {
-    return new Sha256().update(json, 'utf8').digest('hex');
-  }
```

Comment:
Since this is a pure function I think it's a great candidate for extraction, which is perfect because we're injecting this dependency in the tests anyway, and I think it works in production use cases too. It also allows us to remove the Sha256 dependency from this module.